cla_backend/apps/cla_eventlog/migrations/0005_remove_same_day_consecutive_outcome_codes.py

```diff
@@ -0,0 +1,75 @@
+# coding=utf-8
+from __future__ import unicode_literals
+import logging
+from datetime import datetime, time
+from itertools import groupby
+
+from cla_eventlog.constants import LOG_LEVELS, LOG_TYPES
+from django.db import models, migrations
+from django.utils.timezone import now
+from pytz import UTC
+
+logger = logging.getLogger(__name__)
+
+
+def remove_same_day_consecutive_outcome_codes(apps, schema_editor):
+    logger.info('\nLGA-125 data migration: start remove_same_day_consecutive_outcome_codes {}'.format(now()))
+
+    Log = apps.get_model('cla_eventlog', 'Log')
+
+    # Older Django sans TruncDate, etc.
+    outcome_events = Log.objects.filter(type=LOG_TYPES.OUTCOME, level=LOG_LEVELS.HIGH) \
+        .extra(select={'day': 'date( cla_eventlog_log.created )'}) \
+        .values('case__reference', 'code', 'day')
+
+    # First pass. Gather case/day/code with >1 occurrences
+    cases_with_daily_multiple_events = []
+    grouped_results = groupby(outcome_events, key=lambda i: i)
+    for (result, g) in grouped_results:
+        count = sum(1 for _ in g)
+        if count > 1:
+            result['count'] = count
+            cases_with_daily_multiple_events.append(result)
+
+    # Second pass to reverse chronological scan through all log events for a case on a given day
```

Comment:
This is a two-pass clean-up data migration with two for loops. I find it hard to visualise the desired outcome.

Due to the complexity of this code and because it's going to be kept as a data migration, I would feel more confident if it would be unit tested for

- "here's how bad data looks like" -> "here's what we expect it to look like"
- "here's how good data looks like" -> "we expect no changes"

Reply:
Can do, though the desired outcome is super clearly detailed in the attachment to the [originating ticket](https://dsdmoj.atlassian.net/browse/LGA-125).

Can you confirm you've read through `LGA-125 data migration output on staging.txt`? If still not clear, I'll re-frame it with unit tests.

Reply:
As discussed:

- Data migration moved to a management command to separate from deploy
- Log objects serialised in restorable form